src/main/kotlin/no/nav/familie/ef/mottak/repository/domain/Vedlegg.kt

```diff
@@ -15,4 +12,14 @@ data class Vedlegg(@Id
                    val navn: String,
                    val tittel: String,
                    @Convert(converter = FileCryptoConverter::class)
-                   val innhold: Fil)
+                   val innhold: Fil) {
+
+    @PreUpdate private fun preUpdate() {
+        throw UnsupportedOperationException(UPDATE_FEILMELDING)
+    }
+
```

Comment:
Litt magisk for min del, men er dette kode som kjøres før man oppdaterer vedlegget mot i repository-koden? 
Ettersom vi setter id på _alle_ vedlegg - vil den ikke "tro" at alle lagringer er en "merge" (update) ? 🤔 Eller misforstår jeg hvordan rammeverket funker?

Reply:
JPA-magi? Det fungerer ihvertfal i preprodl. Tror bare den kjører på en sql update, ikke en create.

Reply:
Er enig att det blir litt magisk. Den kjører før man oppdaterer vedlegget, så tror det skjer en form av sjekk først som sjekker om id't allerede finnes, hvis det finnes -> update, hvis ikke -> insert

Reply:
```kotlin
//Det som skjer er dette: SimpleJpaRepository
@Transactional
    public <S extends T> S save(S entity) {
        if (this.entityInformation.isNew(entity)) {
            this.em.persist(entity);
            return entity;
        } else {
            return this.em.merge(entity);
        }
    }
```

Reply:
Det er mulig å bruke `: Persistable<UUID>` på Vedlegg, som får en metode som er ìsNew` som alltid returnerer true. Då får man
`Caused by: org.h2.jdbc.JdbcSQLIntegrityConstraintViolationException: Unique index or primary key violation: "PUBLIC.PRIMARY_KEY_3 ON PUBLIC.VEDLEGG(ID) VALUES 1"; SQL statement:` som er det man egentlige forventer.. Men er vel sån ca like weird :P